File: src/models/meal.ts
import type { Food } from "./food";

export type Meal = Readonly<{
  name?: string
  entry?: ReadonlyArray<Food>
  main?: ReadonlyArray<Food>
  side?: ReadonlyArray<Food>
  drink?: ReadonlyArray<Food>
  fromage?: ReadonlyArray<Food>
  dessert?: ReadonlyArray<Food>
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/Meal.kt
package ink.literate.pawnote.models

data class Meal(
    val name: String? = null,
    val entry: List<Food>? = null,
    val main: List<Food>? = null,
    val side: List<Food>? = null,
    val drink: List<Food>? = null,
    val fromage: List<Food>? = null,
    val dessert: List<Food>? = null
)